AdvancedPlaybackSampleApp/app/src/main/java/com/ooyala/sample/players/SetEmbedCodePlayerActivity.java

```diff
@@ -0,0 +1,200 @@
+                player,
+                DefaultControlStyle.AUTO
+        );
+        player.setAdsSeekable(true); // this will help us skip ads if need be.
+        player.addObserver(this);
+        player.setVolume(.05f);
+        int r = player.beginFetchingAdvertisingId(this, this);
+        Log.d( TAG, "initAdvertisingId: " + r );
+    }
+
+    public void onAdvertisingIdSuccess( String adId ) {
+        Log.d( TAG, "onAdvertisingIdSuccess: " + adId );
+    }
+
+    public void onAdvertisingIdError( OoyalaException oe ) {
+        Log.e( TAG, "onAdvertisingIdError", oe );
+    }
+
+    private void setEmbedCode() {
+
+        if (player.setEmbedCode(embedCode)) {
+            player.play();
+            if (embedCode.equals(EMBEDCODE_1)) {
+                embedCode = EMBEDCODE_2;
+            } else {
+                embedCode = EMBEDCODE_1;
+            }
+        } else {
+            Log.d(TAG, "setEmbedCode failed");
+        }
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        if (player != null) {
+            player.suspend();
+        }
+    }
+
+    @Override
+    protected void onRestart() {
+        super.onRestart();
+        if (player != null) {
+            player.resume();
+        }
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        player = null;
+    }
+
+    private final Thread.UncaughtExceptionHandler onUncaughtException = new Thread.UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread thread, Throwable ex) {
+            Log.e(TAG, "Uncaught exception", ex);
+            showErrorDialog(ex);
+        }
+    };
+
+    private void showErrorDialog(Throwable t) {
+        AlertDialog.Builder builder = new AlertDialog.Builder(this);
+
+        builder.setTitle("Exception!");
+        builder.setMessage(t.toString());
+        builder.setPositiveButton("OK", null);
+        builder.show();
+    }
+
+    @Override
+    public void onClick(View arg0) {
+        if (player != null && arg0 == setEmbed) {
+            setEmbedCode();
+        }
+    }
+
+    @Override
+    public void update(Observable arg0, Object arg1) {
+        Log.d(TAG, "Notification Recieved: " + arg1 + " - state: " + player.getState());
+        String notificationName = OoyalaNotification.getNameOrUnknown(arg1);
+        if (notificationName == OoyalaPlayer.CONTENT_TREE_READY_NOTIFICATION_NAME) {
+            Log.d(TAG, "AD - metadata true!");
+        } else if (notificationName == OoyalaPlayer.METADATA_READY_NOTIFICATION_NAME) {
+        } else if (notificationName == OoyalaPlayer.TIME_CHANGED_NOTIFICATION_NAME) {
+        }
+    }
+
+    @Override
+    public void getTokenForEmbedCodes(List<String> embedCodes,
+                                      EmbedTokenGeneratorCallback callback) {
+        String embedCodesString = "";
+        for (String ec : embedCodes) {
+            if(ec.equals("")) embedCodesString += ",";
```

Comment:
please separate this out onto two lines and using braces.

Reply:
done